core/src/main/java/me/lokka30/treasury/plugin/core/command/subcommand/economy/migrate/PlayerAccountMigrationProcess.java

```diff
@@ -0,0 +1,156 @@
+package me.lokka30.treasury.plugin.core.command.subcommand.economy.migrate;
+
+import com.mrivanplays.process.Process;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import me.lokka30.treasury.api.common.response.FailureReason;
+import me.lokka30.treasury.api.common.response.Response;
+import me.lokka30.treasury.api.economy.EconomyProvider;
+import me.lokka30.treasury.api.economy.account.Account;
+import me.lokka30.treasury.api.economy.currency.Currency;
+import me.lokka30.treasury.api.economy.transaction.EconomyTransactionInitiator;
+import org.jetbrains.annotations.NotNull;
+
+class PlayerAccountMigrationProcess extends Process {
+
+    protected final EconomyTransactionInitiator<?> initiator;
+    protected final String accountId;
+    protected final MigrationData migration;
+    protected Account fromAccount, toAccount;
+
+    public PlayerAccountMigrationProcess(
+            @NotNull EconomyTransactionInitiator<?> initiator,
+            @NotNull String accountId,
+            @NotNull MigrationData migration
+    ) {
+        this.initiator = initiator;
+        this.accountId = accountId;
+        this.migration = migration;
+    }
+
+    @Override
+    protected void run() throws Throwable {
+        this.migration.debug(this::getInitLog);
+
+        // Retrieve accounts
+        Response<Account> fromAccountResp = this.requestOrCreateAccount(this.migration.from(),
+                this.accountId
+        ).get();
+        Response<Account> toAccountResp = this.requestOrCreateAccount(this.migration.to(),
+                this.accountId
+        ).get();
+
+        if (!handleUnsuccessfulResponse(fromAccountResp) || !handleUnsuccessfulResponse(
+                toAccountResp)) {
+            return;
+        }
+
+        this.fromAccount = fromAccountResp.getResult();
+        this.toAccount = toAccountResp.getResult();
+
+        // retrieve currencies
+        Response<Collection<String>> heldCurrenciesResp = fromAccount
+                .retrieveHeldCurrencies()
+                .get();
+        if (!handleUnsuccessfulResponse(heldCurrenciesResp)) {
+            return;
+        }
+
+        for (String id : heldCurrenciesResp.getResult()) {
+            // find currency in the new provider
+            Optional<Currency> currencyOpt = this.migration.to().findCurrency(id);
+            Currency currency;
+            if (currencyOpt.isPresent()) {
+                currency = currencyOpt.get();
+            } else {
+                // TODO: perhaps create the currency in the new provider?
```

Comment:
shall we implement this? the original v1 code did not.

Reply:
I think there initially wasn't a way to create a new currency via the API, but I could be wrong. The v1 migrator pre-mapped currencies, and that would imo be an appropriate way to create currencies if we do plan to map them. That way all currencies, not just ones currently held by players, would be migrated.